java/google/registry/rde/imports/XjcToDomainResourceConverter.java

```diff
@@ -293,16 +238,34 @@ private static HistoryEntry createHistoryEntry(XjcRdeDomain domain) {
   }
 
   /** Converts {@link XjcRdeDomainTransferDataType} to {@link TransferData}. */
-  private static TransferData convertDomainTransferData(XjcRdeDomainTransferDataType data) {
+  private static TransferData convertDomainTransferData(XjcRdeDomainTransferDataType data,
+      DateTime domainExpiration) {
     if (data == null) {
       return TransferData.EMPTY;
     }
+    // If the transfer is pending, calculate the number of years to add to the domain expiration
+    // on approval of the transfer.
+    TransferStatus transferStatus = TRANSFER_STATUS_MAPPER.xmlToEnum(data.getTrStatus().value());
+    // Get new expiration date
+    DateTime newExpirationTime = domainExpiration;
+    if (transferStatus == TransferStatus.PENDING) {
+      // Default to domain expiration time plus one year if no expiration is specified
+      if (data.getExDate() == null) {
+        newExpirationTime = newExpirationTime.plusYears(1);
+      } else {
+        newExpirationTime = data.getExDate();
+      }
+    }
     return new TransferData.Builder()
-        .setTransferStatus(TRANSFER_STATUS_MAPPER.xmlToEnum(data.getTrStatus().value()))
+        .setTransferStatus(transferStatus)
         .setGainingClientId(data.getReRr().getValue())
         .setLosingClientId(data.getAcRr().getValue())
         .setTransferRequestTime(data.getReDate())
         .setPendingTransferExpirationTime(data.getAcDate())
+        // This will be wrong for domains that are not in pending transfer,
```

Comment:
Our extendedRegistrationYears field has a number of problems overall, not the least of which is that as far as I can tell, ICANN policy stipulates that transfers must include a fixed 1-year registration extension, not an arbitrary number of years.  See section A.8 of ICANN's ["Policy on Transfer of Registrations between Registrars"](https://www.icann.org/resources/pages/policy-2012-03-07-en) (note that regular transfers are covered under section A, "Holder-authorized transfers"; section B handles "ICANN-Approved transfers" which are bulk transfers between one registrar to another, e.g. upon going out of business or being acquired).

> 8. Effect on Term of Registration
> The completion by Registry Operator of a holder-authorized transfer under this Part A shall result in a one-year extension of the existing registration, provided that in no event shall the total unexpired term of a registration exceed ten (10) years.

At least, that's the interpretation that seems most straightforward.  While EPP in general and our registry implementation to date allow specifying a number >1 of years to renew in a transfer (see [the tests](https://github.com/google/nomulus/blob/master/javatests/google/registry/flows/domain/testdata/domain_transfer_request_response_10_years.xml)), evidently that's not the same as the policy.  And in practice it appears that registrars only do 1-year transfers.

So I'm hoping to fix that and in the process eliminate this field, which will eliminate this issue.

However, all that said, I'm pointing out just FYI that this calculation isn't even always correct for domains that are in pending transfer, due to 10-year capping.  As mentioned in my other comment, if the transfer would hit 10-year capping (e.g. the domain was created w/ a 10 year registration, next day a transfer is requested w/ bundled 1-year registration extension), then exDate won't necessarily be a full year from the current expiration time, and the Years.yearsBetween() call will truncate and return 0, which is never a valid value of extendedRegistrationYears.